cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -92,7 +97,13 @@ public BasicMapReduceContext(Program program,
     } else {
       this.userMetrics = null;
     }
-    this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    String adapterName = getAdapterName(runtimeArguments);
+    if (adapterName == null) {
+      this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    } else {
+      this.loggingContext = new BatchAdapterLoggingContext(getNamespaceId(), getApplicationId(), getProgramName(),
```

Comment:
Alternatively, do the extraction at the AbstractContext level so that it can set for both metrics and logging